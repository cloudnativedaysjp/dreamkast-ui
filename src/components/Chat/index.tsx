export { Chat } from './Chat'
import { ChatMessage as ChatMessageInterface, ChatMessageMessageTypeEnum } from '../../client-axios'


export class ChatMessageClass implements ChatMessageInterface {
  id?: number;
  profileId? :number;
  speakerId?: number;
  eventAbbr: string;
  roomId?: number;
  roomType?: string;
  body: string;
  messageType: ChatMessageMessageTypeEnum;

  constructor(id: number, profileId: number, speakerId: number, eventAbbr: string, roomId: number, roomType: string, body: string, messageType: ChatMessageMessageTypeEnum) {
    this.id = id;
    this.profileId = profileId;
    this.speakerId = speakerId;
    this.eventAbbr = eventAbbr;
    this.roomId = roomId;
    this.roomType = roomType;
    this.body = body;
    this.messageType = messageType
  }

  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
  createFromWebSocket(
    id: number,
    eventAbbr: string,
    roomId: number,
    roomType: string,
    body: string,
  ) {
    this.id = id
    this.eventAbbr = eventAbbr
    this.roomId = roomId
    this.roomType = roomType
    this.body = body
  }
}
