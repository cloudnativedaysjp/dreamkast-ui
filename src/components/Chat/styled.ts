import styled from 'styled-components'
import { Paper, Box as MaterialUIBox } from '@material-ui/core'

export const Box = styled(MaterialUIBox)`
  height: 400px;
`

export const ChatMessage = styled(Paper)<{ isChat: boolean }>`
  padding: 10px;
  background: ${({ isChat }) => (isChat ? '#ffffff' : '#f5f5f5')};
  white-sace: pre-wrap;
`
export const ChatReplyMessage = styled(ChatMessage)`
  margin-left: 20px;
`

export const ChatSelectedMessage = styled(ChatMessage)`
  background-color: cornsilk;
`

export const CloseReplyButton = styled.div`
  cursor: pointer;
  float: right;
  margintop: 5px;
  width: 20px;
`
