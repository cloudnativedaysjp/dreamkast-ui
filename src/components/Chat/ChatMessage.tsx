import React, { useEffect, useRef } from 'react'
import * as Styled from './styled'
import {
  ChatMessage as ChatMessageObject,
  ChatMessageMessageTypeEnum,
} from '../../client-axios/api'

type Props = {
  chatMessage: ChatMessageObject
  setRef: (
    chatMessage: ChatMessageObject,
    ref: React.RefObject<HTMLDivElement>,
  ) => void
}

const ChatMessage: React.FC<Props> = ({ chatMessage, setRef }) => {
  const ref = useRef<HTMLDivElement>(null)
  const isSpeakerMessage = !!chatMessage.speaker_id

  useEffect(() => {
    setRef(chatMessage, ref)
  }, [setRef, chatMessage, ref])

  return (
    <div ref={ref}>
      <Styled.ChatMessage
        isChat={chatMessage.messageType == ChatMessageMessageTypeEnum.Chat}
      >
        {isSpeakerMessage ? '[S] ' : ''}
        {chatMessage.body}
      </Styled.ChatMessage>
    </div>
  )
}

export default ChatMessage
