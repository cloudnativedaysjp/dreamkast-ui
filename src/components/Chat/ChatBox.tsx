import React from 'react'
import { Box} from '@material-ui/core'
import { ChatMessage as ChatMessageInterface, ChatMessageMessageTypeEnum } from '../../client-axios'
import {makeStyles} from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import ChatMessage from "./ChatMessage";

type Props = {
  messages: ChatMessageInterface[]
  messageTypes: ChatMessageMessageTypeEnum[]
}
const useStyles = makeStyles((theme) => ({
    box: {
        height: '400px',
    },
    paper: {
        padding: theme.spacing(2),
        color: theme.palette.text.secondary,
    },
}))

export const ChatBox: React.FC<Props> = ({ messages, messageTypes }) => {
  const classes = useStyles()
  const setLastMessageElement = (
    chatMessage: ChatMessageInterface,
    ref: React.RefObject<HTMLDivElement>,
  ) => {
    const lastChat = messages[messages.length - 1]
    if (chatMessage.id === lastChat.id) {
      ref && ref.current && ref.current.scrollIntoView()
    }
  }

  return (
    <Box component="div" overflow="scroll" className={classes.box}>
      <Grid container spacing={3}>
        <Grid item xs={12}>
          {messages.map((chatMessage) => {
            if (messageTypes.includes(chatMessage.messageType)) {
              return (
                <ChatMessage
                  chatMessage={chatMessage}
                  setRef={setLastMessageElement}
                />
              )
            }
          })}
        </Grid>
      </Grid>
    </Box>
  )
}
