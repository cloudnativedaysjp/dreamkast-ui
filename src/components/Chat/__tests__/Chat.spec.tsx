import React from 'react'
import 'cross-fetch/polyfill'
import { rest } from 'msw'
import { Chat } from '../Chat'
import { renderWithProviders, setupStore } from '../../../testhelper/store'
import {
  MockChats,
  MockEvent,
  MockProfile,
  MockTalkA1,
} from '../../../testhelper/fixture'
import { GetApiV1ChatMessagesApiResponse } from '../../../generated/dreamkast-api.generated'
import { setupMockServer } from '../../../testhelper/msw'
import { setProfile } from '../../../store/settings'
import { setWsBaseUrl } from '../../../store/auth'
import { fireEvent } from '@testing-library/react'

const server = setupMockServer(
  rest.get(`/api/v1/chat_messages`, (_, res, ctx) => {
    return res(ctx.json(MockChats() as GetApiV1ChatMessagesApiResponse))
  }),
  rest.get('http://localhost:8080/cable', (_, res, ctx) => {
    return res(ctx.status(101))
  }),
)

describe('Chat', () => {
  it('should fetch chat data and render without crash', async () => {
    const got = {
      eventAbbr: '',
      roomId: '',
      roomType: '',
    }

    const fn = jest.fn()
    const fnCable = jest.fn()
    server.use(
      rest.get(`/api/v1/chat_messages`, (req, res, ctx) => {
        got.eventAbbr = req.url.searchParams.get('eventAbbr') as string
        got.roomId = req.url.searchParams.get('roomId') as string
        got.roomType = req.url.searchParams.get('roomType') as string
        fn()
        return res(ctx.json(MockChats() as GetApiV1ChatMessagesApiResponse))
      }),
      rest.get('http://localhost:8080/cable', (_, res, ctx) => {
        fnCable()
        return res(ctx.status(101))
      }),
    )

    const mockProps = {
      event: MockEvent(),
      talk: MockTalkA1(),
    }

    const store = setupStore()
    store.dispatch(setProfile(MockProfile()))
    store.dispatch(setWsBaseUrl('http://localhost:8080'))
    const screen = renderWithProviders(<Chat {...mockProps} />, { store })
    const want = {
      eventAbbr: mockProps.event.abbr,
      roomId: `${mockProps.talk.id}`,
      roomType: 'talk',
    }

    await screen.findAllByText('わいわい')
    expect(fn).toHaveBeenCalled()
    expect(fnCable).toHaveBeenCalled()
    expect(got).toStrictEqual(want)
    expect(screen.asFragment()).toMatchSnapshot()
  })

  it('should render without crash when no data provided', async () => {
    const fn = jest.fn()
    const fnCable = jest.fn()
    server.use(
      rest.get(`/api/v1/chat_messages`, (_, res, ctx) => {
        fn()
        return res(ctx.json([] as GetApiV1ChatMessagesApiResponse))
      }),
      rest.get('http://localhost:8080/cable', (_, res, ctx) => {
        fnCable()
        return res(ctx.status(101))
      }),
    )

    const mockProps = {
      event: MockEvent(),
    }

    const store = setupStore()
    store.dispatch(setProfile(MockProfile()))
    store.dispatch(setWsBaseUrl('http://localhost:8080'))
    const screen = renderWithProviders(<Chat {...mockProps} />, { store })

    expect(fn).not.toHaveBeenCalled()
    expect(fnCable).not.toHaveBeenCalled()
    expect(screen.asFragment()).toMatchSnapshot()
  })

  it('should create new chat', async () => {
    let called = false
    server.use(
      rest.post('/api/v1/chat_messages', (_, res) => {
        called = true
        return res()
      }),
    )

    const mockProps = {
      event: MockEvent(),
      talk: MockTalkA1(),
    }

    const store = setupStore()
    store.dispatch(setProfile(MockProfile()))
    store.dispatch(setWsBaseUrl('http://localhost:8080'))
    const screen = renderWithProviders(<Chat {...mockProps} />, { store })
    await screen.findAllByText('わいわい')

    fireEvent.change(screen.getByTestId('message-form'), {
      target: { value: 'テスト' },
    })
    fireEvent.click(screen.getByTestId('submit-chat'))

    const ok = await new Promise((resolve) => {
      let count = 0
      setInterval(() => {
        if (called) {
          resolve(true)
        }
        count++
        if (count > 10) {
          resolve(false)
        }
      }, 10)
    })
    expect(ok).toBeTruthy()
  })
})
