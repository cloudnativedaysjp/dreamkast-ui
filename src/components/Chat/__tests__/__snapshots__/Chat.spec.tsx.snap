// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat should fetch chat data and render without crash 1`] = `
<DocumentFragment>
  <div
    class="styled__Container-sc-1llqfms-0 bBAkaJ"
  >
    <div
      class="MuiTabs-root styled__TabContainer-sc-1llqfms-1 dJrtFc"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed"
        style="overflow: hidden;"
      >
        <div
          aria-label="simple tabs example"
          class="MuiTabs-flexContainer"
          role="tablist"
        >
          <button
            aria-controls="simple-tabpanel-0"
            aria-labelledby="simple-tabpanel-0"
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary styled__Tab-sc-1llqfms-3 BXQwU Mui-selected"
            id="simple-tab-0"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              Chat / QA
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-controls="simple-tabpanel-1"
            aria-labelledby="simple-tabpanel-1"
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary styled__Tab-sc-1llqfms-3 BXQwU"
            id="simple-tab-1"
            role="tab"
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              QA Only
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <span
          class="PrivateTabIndicator-root-1 PrivateTabIndicator-colorSecondary-3 MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
    <div
      aria-controls="simple-tabpanel-0"
      aria-labelledby="simple-tabpanel-0"
      class="MuiTabPanel-root styled__TabPanel-sc-1llqfms-2 ljDeDA"
      id="simple-tab-0"
      role="tabpanel"
    >
      <div
        class="MuiBox-root MuiBox-root-5 styled__Box-sc-7eb3vm-0 kkjyek"
        id="message-box"
      >
        <div>
          <div
            class="MuiPaper-root styled__ChatMessage-sc-h0bng9-0 cCTQBk MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11"
              >
                08:41
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
              >
                <div
                  class="styled__MenuButton-sc-h0bng9-3 RvpXS"
                  data-messageid="286"
                  data-testid="message-menu-btn"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root styled__MenuIcon-sc-h0bng9-4 ieQDrx MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11"
              >
                <div
                  class="styled__MessageBody-sc-h0bng9-1 mlXih"
                >
                  „Çè„ÅÑ„Çè„ÅÑ
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
              >
                <div
                  class="styled__ReplyButton-sc-h0bng9-5 ilnbPt"
                  data-messageid="286"
                  data-testid="message-reply-btn"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root styled__ReplyIcon-sc-h0bng9-6 hpRTLe MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 9V5l-7 7 7 7v-4.1c5 0 8.5 1.6 11 5.1-1-5-4-10-11-11z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="MuiPaper-root styled__ChatMessage-sc-h0bng9-0 cCTQBk MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11"
              >
                11:27
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
              >
                <div
                  class="styled__MenuButton-sc-h0bng9-3 RvpXS"
                  data-messageid="288"
                  data-testid="message-menu-btn"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root styled__MenuIcon-sc-h0bng9-4 ieQDrx MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11"
              >
                <div
                  class="styled__MessageBody-sc-h0bng9-1 mlXih"
                >
                  „Çè„ÅÑ„Çè„ÅÑ
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
              >
                <div
                  class="styled__ReplyButton-sc-h0bng9-5 ilnbPt"
                  data-messageid="288"
                  data-testid="message-reply-btn"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root styled__ReplyIcon-sc-h0bng9-6 hpRTLe MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 9V5l-7 7 7 7v-4.1c5 0 8.5 1.6 11 5.1-1-5-4-10-11-11z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiPaper-root styled__ChatMessage-sc-h0bng9-0 styled__ChatReplyMessage-sc-h0bng9-2 cCTQBk loaNci MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="styled__MessageBody-sc-h0bng9-1 mlXih"
            >
              „Åå„ÇÑ„Åå„ÇÑ
            </div>
            <div
              class="styled__MenuButton-sc-h0bng9-3 RvpXS"
              data-messageid="343"
              data-replyto="288"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiPaper-root styled__ChatMessage-sc-h0bng9-0 styled__ChatReplyMessage-sc-h0bng9-2 cCTQBk loaNci MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="styled__MessageBody-sc-h0bng9-1 mlXih"
            >
              üëç
            </div>
            <div
              class="styled__MenuButton-sc-h0bng9-3 RvpXS"
              data-messageid="344"
              data-replyto="288"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiPaper-root styled__ChatMessage-sc-h0bng9-0 styled__ChatReplyMessage-sc-h0bng9-2 cCTQBk loaNci MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="styled__MessageBody-sc-h0bng9-1 mlXih"
            >
              „Åì„ÅÆ„É°„ÉÉ„Çª„Éº„Ç∏„ÅØÂâäÈô§„Åï„Çå„Åæ„Åó„Åü
            </div>
            <div
              class="styled__MenuButton-sc-h0bng9-3 RvpXS"
              data-messageid="345"
              data-replyto="288"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div>
          <div
            class="MuiPaper-root styled__ChatMessage-sc-h0bng9-0 ccGrOO MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11"
              >
                11:26
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
              >
                <div
                  class="styled__MenuButton-sc-h0bng9-3 RvpXS"
                  data-messageid="289"
                  data-testid="message-menu-btn"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root styled__MenuIcon-sc-h0bng9-4 ieQDrx MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11"
              >
                <div
                  class="styled__MessageBody-sc-h0bng9-1 mlXih"
                >
                  „Åì„ÅÆ„É°„ÉÉ„Çª„Éº„Ç∏„ÅØÂâäÈô§„Åï„Çå„Åæ„Åó„Åü
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
              />
            </div>
          </div>
        </div>
        <div>
          <div
            class="MuiPaper-root styled__ChatMessage-sc-h0bng9-0 cCTQBk MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11"
              >
                13:40
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
              >
                <div
                  class="styled__MenuButton-sc-h0bng9-3 RvpXS"
                  data-messageid="346"
                  data-testid="message-menu-btn"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root styled__MenuIcon-sc-h0bng9-4 ieQDrx MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11"
              >
                <div
                  class="styled__MessageBody-sc-h0bng9-1 mlXih"
                >
                  üéâ
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
              >
                <div
                  class="styled__ReplyButton-sc-h0bng9-5 ilnbPt"
                  data-messageid="346"
                  data-testid="message-reply-btn"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root styled__ReplyIcon-sc-h0bng9-6 hpRTLe MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 9V5l-7 7 7 7v-4.1c5 0 8.5 1.6 11 5.1-1-5-4-10-11-11z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-controls="simple-tabpanel-1"
      aria-labelledby="simple-tabpanel-1"
      class="MuiTabPanel-root styled__TabPanel-sc-1llqfms-2 ljDeDA"
      hidden=""
      id="simple-tab-1"
      role="tabpanel"
    />
    <div
      class="styled__Container-sc-nbl6ba-0 cZiACc"
    >
      <form
        class="styled__ChatMessageForm-sc-nbl6ba-5"
      >
        <div
          class="styled__CheckBoxContainer-sc-nbl6ba-9 khgagE"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-7 Mui-checked MuiIconButton-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input-9"
                data-indeterminate="false"
                type="checkbox"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <div
            class="styled__label-sc-nbl6ba-10"
          >
            „Ç™„Éº„Éà„Çπ„ÇØ„É≠„Éº„É´
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root styled__TextField-sc-nbl6ba-7 kRBvPZ"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorSecondary MuiInput-colorSecondary MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
              data-testid="message-form"
              name="chatMessage"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline"
        >
          <input
            class="MuiInputBase-input MuiInput-input"
            name="isQuestion"
            type="hidden"
            value=""
          />
        </div>
        <div
          class="styled__ButtonContainer-sc-nbl6ba-6 eVOFBv"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline"
          >
            <input
              class="MuiInputBase-input MuiInput-input"
              name="chatMessage"
              type="hidden"
              value="üëç"
            />
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained styled__ReactButton-sc-158ay7x-1 fhKoAh"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              üëç
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <form
            class="styled__ChatMessageForm-sc-158ay7x-0"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline"
            >
              <input
                class="MuiInputBase-input MuiInput-input"
                name="chatMessage"
                type="hidden"
                value="üëè"
              />
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained styled__ReactButton-sc-158ay7x-1 fhKoAh"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                üëè
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </form>
          <form
            class="styled__ChatMessageForm-sc-158ay7x-0"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline"
            >
              <input
                class="MuiInputBase-input MuiInput-input"
                name="chatMessage"
                type="hidden"
                value="üéâ"
              />
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained styled__ReactButton-sc-158ay7x-1 fhKoAh"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                üéâ
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </form>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained Mui-disabled Mui-disabled"
            data-testid="submit-chat"
            disabled=""
            tabindex="-1"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              ÈÄÅ‰ø°
            </span>
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained Mui-disabled Mui-disabled"
            data-testid="submit-qa"
            disabled=""
            tabindex="-1"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Ë≥™Âïè„Åô„Çã
            </span>
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Chat should render without crash when no data provided 1`] = `
<DocumentFragment>
  <div
    class="styled__Container-sc-1llqfms-0 bBAkaJ"
  >
    <div
      class="MuiTabs-root styled__TabContainer-sc-1llqfms-1 dJrtFc"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed"
        style="overflow: hidden;"
      >
        <div
          aria-label="simple tabs example"
          class="MuiTabs-flexContainer"
          role="tablist"
        >
          <button
            aria-controls="simple-tabpanel-0"
            aria-labelledby="simple-tabpanel-0"
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary styled__Tab-sc-1llqfms-3 BXQwU Mui-selected"
            id="simple-tab-0"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              Chat / QA
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-controls="simple-tabpanel-1"
            aria-labelledby="simple-tabpanel-1"
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary styled__Tab-sc-1llqfms-3 BXQwU"
            id="simple-tab-1"
            role="tab"
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              QA Only
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <span
          class="PrivateTabIndicator-root-10 PrivateTabIndicator-colorSecondary-12 MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
    <div
      aria-controls="simple-tabpanel-0"
      aria-labelledby="simple-tabpanel-0"
      class="MuiTabPanel-root styled__TabPanel-sc-1llqfms-2 ljDeDA"
      id="simple-tab-0"
      role="tabpanel"
    >
      <div
        class="MuiBox-root MuiBox-root-14 styled__Box-sc-7eb3vm-0 kkjyek"
        id="message-box"
      />
    </div>
    <div
      aria-controls="simple-tabpanel-1"
      aria-labelledby="simple-tabpanel-1"
      class="MuiTabPanel-root styled__TabPanel-sc-1llqfms-2 ljDeDA"
      hidden=""
      id="simple-tab-1"
      role="tabpanel"
    />
    <div
      class="styled__Container-sc-nbl6ba-0 cZiACc"
    >
      <form
        class="styled__ChatMessageForm-sc-nbl6ba-5"
      >
        <div
          class="styled__CheckBoxContainer-sc-nbl6ba-9 khgagE"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-15 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-16 Mui-checked MuiIconButton-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input-18"
                data-indeterminate="false"
                type="checkbox"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <div
            class="styled__label-sc-nbl6ba-10"
          >
            „Ç™„Éº„Éà„Çπ„ÇØ„É≠„Éº„É´
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root styled__TextField-sc-nbl6ba-7 kRBvPZ"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorSecondary MuiInput-colorSecondary MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
              data-testid="message-form"
              name="chatMessage"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline"
        >
          <input
            class="MuiInputBase-input MuiInput-input"
            name="isQuestion"
            type="hidden"
            value=""
          />
        </div>
        <div
          class="styled__ButtonContainer-sc-nbl6ba-6 eVOFBv"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline"
          >
            <input
              class="MuiInputBase-input MuiInput-input"
              name="chatMessage"
              type="hidden"
              value="üëç"
            />
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained styled__ReactButton-sc-158ay7x-1 fhKoAh"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              üëç
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <form
            class="styled__ChatMessageForm-sc-158ay7x-0"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline"
            >
              <input
                class="MuiInputBase-input MuiInput-input"
                name="chatMessage"
                type="hidden"
                value="üëè"
              />
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained styled__ReactButton-sc-158ay7x-1 fhKoAh"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                üëè
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </form>
          <form
            class="styled__ChatMessageForm-sc-158ay7x-0"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline"
            >
              <input
                class="MuiInputBase-input MuiInput-input"
                name="chatMessage"
                type="hidden"
                value="üéâ"
              />
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained styled__ReactButton-sc-158ay7x-1 fhKoAh"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                üéâ
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </form>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained Mui-disabled Mui-disabled"
            data-testid="submit-chat"
            disabled=""
            tabindex="-1"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              ÈÄÅ‰ø°
            </span>
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained Mui-disabled Mui-disabled"
            data-testid="submit-qa"
            disabled=""
            tabindex="-1"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Ë≥™Âïè„Åô„Çã
            </span>
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
